### PR TITLE
Rename example to simple-native + update mdbook + add markdownlint CI

### PR DESCRIPTION
**Pull request by ra0x3**
### Description
- This PR does a few things:
  - Renames the example from simple-non-wasm to simple-native
  - Adds `markdownlint` to CI for linting the mdbook
  - Updates the mdbook simple-native example to match work done in https://github.com/FuelLabs/fuel-indexer/pull/90

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -214,6 +214,16 @@ jobs:
...
 
+  mdbook-lint:
+    needs: cancel-previous-runs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: npm install -g markdownlint-cli && markdownlint --config .markdownlintrc **/*.md docs/src/
```
Comment (adlerjohn) [inline .github/workflows/ci.yml:225]:
```suggestion
      - run: |
          npm install -g markdownlint-cli
          markdownlint --config .markdownlintrc **/*.md docs/src/ 
```

Comment (adlerjohn) [inline .github/workflows/ci.yml:225]: I would prefer if an exact version of markdownlint-cli were installed. I've had too many nightmares of JS dependencies getting bitrot and suddenly stop working 😂

Comment (ra0x3) [reply .github/workflows/ci.yml]: That's fair. Updates pushed

File: docs/src/examples/simple-native.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+```
+
+### Start the fuel indexer service
+
+With our Fuel node and webserver up and running, we'll next start our Fuel Indexer service.
+
+```bash
+cd fuel-indexer/examples/simple-native/
+
+RUST_LOG=info cargo run -- ./config.yaml ./manifest.yaml
```
Comment (mitchmindtree) [inline docs/src/examples/simple-native.md:100]:
```suggestion
cargo run -- ./config.yaml ./manifest.yaml
```

I could be wrong, but I *think* `info` is the default log level?

Also, it's a shame we have to pass `./config.yaml` and `./manifest.yaml` manually, but I guess working out some sort of standard/default locations for these things could be worked out as a part of #89 or some other issue.

**Review by mitchmindtree — COMMENTED** [review]
Nice one @ra0x3, looking good :ok: Just a couple nits

File: docs/src/examples/simple-native.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+- Note that some of these commands may differ based on your local setup
+
+```bash
+createdb -U postgres indexer
+```
+
+Next we'll bootstrap our database by running some migrations. These migrations are repsonsible for creating our `graph_registry`. 
+
+- The `graph_registry` is a component that is responsible for keeping track of database columns and columns types. The `graph_registry` is largely abstracted away from the user, but can be seen by inspecting the `graph_registry` schema.
```
[the remark this answers is not in the document]
Comment (adlerjohn) [reply docs/src/examples/simple-native.md:74]: You can also use 

```markdown
> **Note**
> text
```

or

```markdown
> **Note** text
```

and it'll render with a cute `Note (!)` icon on both GitHub and mdbook.

[the remark this answers is not in the document]
Comment (ra0x3) [reply docs/src/examples/simple-native.md:74]: @mitchmindtree 
- I'm down for that. Unfortunate that there's a codeblock just underneath, and these quote boxes look....just like code blocks 😆 🤦 
- Will push a change

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -214,6 +214,18 @@ jobs:
...
+  mdbook-lint:
+    needs: cancel-previous-runs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: |
+          npm install -g markdownlint-cli@https://github.com/igorshubovych/markdownlint-cli/archive/refs/tags/v0.31.1.tar.gz
```
Comment (adlerjohn) [inline .github/workflows/ci.yml:226]: Does

```suggestion
          npm install -g markdownlint-cli@0.31.1
```

not work?

Comment (ra0x3) [reply .github/workflows/ci.yml]: @adlerjohn It most certainly does (I think this is a holdover from some debugging I was doing). Change pushed

**Review by JoshuaBatty — APPROVED** [review]

**Pull request merged**